### PR TITLE
Compile and use jpsreport on windows

### PR DESCRIPTION
**Pull request by xuqiancheng**
- ba4e776 Fix some problems when compiling jpsreport on windows with VS2017. Problem related with OpenMP and link with stdc++fs( it seems that we dont't have to link stdc++fs when compile with MSVC).

- 937c3c6 Since the difference between '/' and '\\', there are some errors when using jpsreport on windows. I have fixed these errors, now it works good on my computer, but the code need to be tested on another windows and on other OS.

### CODE REVIEW

**Comment by gjaeger** [comment]
- ba4e776 I don't get an error message (see [cmake_protocol.log](https://github.com/JuPedSim/jpsreport/files/3491125/cmake_protocol.log)). We should always check the cmake file, like jpscore.
- 937c3c6 I get your changes now as a warning (see [make_protocol.log](https://github.com/JuPedSim/jpsreport/files/3491132/make_protocol.log)). We should not only solve this problem for one operating system.

File: Analysis.cpp [hunk 1]
```diff
@@ -546,17 +546,19 @@ int Analysis::mkpath(char* file_path)
 {
      assert(file_path && *file_path);
      char* p;
-     for (p=strchr(file_path+1, '/'); p; p=strchr(p+1, '/')) {
+	 for (p = strchr(file_path + 1, '/'); p; p = strchr(p + 1, '/')) {
```
Comment (schroedtert) [inline Analysis.cpp]: As later on filesystem::path is used, is this working on the filenames needed?

File: general/ArgumentParser.cpp [hunk 2]
```diff
@@ -352,6 +352,9 @@ bool ArgumentParser::ParseIniFile(const string& inifile)
                     for (auto& filename : boost::make_iterator_range(fs::directory_iterator(path_traj), {}))
                     {
                          string s = filename.path().string();
+						 string::size_type pos = s.find_last_of('/');
```
Comment (schroedtert) [inline general/ArgumentParser.cpp]: Too many blanks

Comment (xuqiancheng) [reply general/ArgumentParser.cpp]: > Too many blanks

I always forget to use space instead of tab, Sorry for this. 
I have fixed it in  42937b3

**Comment by xuqiancheng** [comment]
@gjaeger 
Yes, I agree with you. We should check the cmake file, like jpscore.
I have made some mistakes, 42937b3 fixed it. And I have tested it on manjaro, it works without warning.  Could you test it on your OS, thanks.

File: Analysis.cpp [hunk 3]
```diff
@@ -76,6 +76,7 @@ Analysis::Analysis()
      _DoesUseMethodB = false;                   // Method B (Zhang2011a)
      _DoesUseMethodC = false;                                   // Method C //calculate and save results of classic in separate file
      _DoesUseMethodD = false;                                   // Method D--Voronoi method
+     _DoesUseMethodI = false;
```
Comment (gjaeger) [inline Analysis.cpp]: Thank you very much, we forgot that during the implementation.

File: Analysis.cpp (hunk 1 above)
Comment (chraibi) [reply Analysis.cpp]: just remove this function, please. It's old and probably unused.

Comment (xuqiancheng) [reply Analysis.cpp]: > just remove this function, please. It's old and probably unused.

This function is used to create directories for output. I am not familiar with filesystem::path, is there any new function can realize it easier?

Comment (chraibi) [reply Analysis.cpp]: Yes, there is a function 
https://en.cppreference.com/w/cpp/filesystem/create_directory
I see now, that even ` Analysis::CreateFile()` is not necessary, maybe you can check this answer here
https://stackoverflow.com/a/53356501

Comment (xuqiancheng) [reply Analysis.cpp]: > Yes, there is a function
> https://en.cppreference.com/w/cpp/filesystem/create_directory
> I see now, that even ` Analysis::CreateFile()` is not necessary, maybe you can check this answer here
> https://stackoverflow.com/a/53356501

See commit 41c0d9b.

**Comment by chraibi** [comment]
I can compile this and run basic tests. It looks good. 
@gjaeger  unless you have any concerns, I think this should be accepted and merged.

**Comment by gjaeger** [comment]
No concerns. We can merge.

**Pull request merged**